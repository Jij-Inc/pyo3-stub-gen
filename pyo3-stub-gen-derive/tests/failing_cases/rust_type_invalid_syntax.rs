use pyo3_stub_gen_derive::gen_function_from_python;

fn main() {
    pyo3_stub_gen::inventory::submit! {
        gen_function_from_python! {
            r#"
            def process(x: pyo3_stub_gen.RustType["Invalid:::"]) -> int:
                """Function using invalid Rust type syntax"""
            "#
        }
    }
}
