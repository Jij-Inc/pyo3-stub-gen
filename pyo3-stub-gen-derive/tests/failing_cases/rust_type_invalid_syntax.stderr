error: Failed to parse Rust type 'Invalid:::': expected identifier
  --> tests/failing_cases/rust_type_invalid_syntax.rs:5:9
   |
 5 | /         gen_function_from_python! {
 6 | |             r#"
 7 | |             def process(x: pyo3_stub_gen.RustType["Invalid:::"]) -> int:
 8 | |                 """Function using invalid Rust type syntax"""
 9 | |             "#
10 | |         }
   | |_________^
   |
   = note: this error originates in the macro `gen_function_from_python` (in Nightly builds, run with -Z macro-backtrace for more info)
